open_flamingo/eval/evaluate.py

```diff
@@ -296,94 +344,98 @@ def get_random_indices(num_samples, query_set_size, full_dataset, seed):
     return random_indices
 
 
-def prepare_eval_samples_and_dataset(full_dataset, random_indices, query_set_size):
-    # get in context samples
-    in_context_samples = [full_dataset[i] for i in random_indices[:query_set_size]]
-    eval_dataset = torch.utils.data.Subset(
-        full_dataset, random_indices[query_set_size:]
-    )
-    return in_context_samples, eval_dataset
-
-
-def get_context_text(
-    get_prompt: Callable[[dict], str],
-    in_context_samples,
-    effective_num_shots,
-    num_shots,
-) -> str:
-    context_text = (
-        "".join([get_prompt(s) for s in in_context_samples])
-        if effective_num_shots > 0
-        else ""
-    )
+def get_query_set(train_dataset, query_set_size, seed):
+    np.random.seed(seed)
+    query_set = np.random.choice(len(train_dataset), query_set_size, replace=False)
+    return [train_dataset[i] for i in query_set]
 
-    if num_shots == 0:
-        context_text = context_text.replace("<image>", "")
-    return context_text
+
+def prepare_eval_samples(test_dataset, num_samples, seed):
+    np.random.seed(seed)
+    random_indices = np.random.choice(len(test_dataset), num_samples, replace=False)
+    return torch.utils.data.Subset(test_dataset, random_indices)
 
 
 def sample_batch_demos_from_query_set(query_set, num_samples, batch_size):
     return [random.sample(query_set, num_samples) for _ in range(batch_size)]
 
 
-def evaluate_coco_flickr(
+def evaluate_captioning(
+    args,
     eval_model,
-    batch_size,
-    image_dir_path,
-    annotations_json_path,
     seed=42,
     max_generation_length=20,
```

Comment:
Are these standard choices for hyperparameters? will we ever need longer generations? thinking maybe this should be a flag.